File: app/src/main/java/com/hypnoweb/hypnowebapp/catalog/repo/network/CatalogNetworkMapper.java
package com.hypnoweb.hypnowebapp.catalog.repo.network;

import com.hypnoweb.hypnowebapp.catalog.repo.Catalog;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.Single;

class CatalogNetworkMapper {

    @Inject
    public CatalogNetworkMapper() {
    }

    public Single<Catalog> fromNetworkCatalog(CatalogNetwork catalogNetwork) {
        return Observable.just(catalogNetwork.getMovies())
                .flatMapIterable(moviesNetwork -> moviesNetwork)
                .map(movieNetwork -> new Catalog.Movie(movieNetwork.getMovieId(), movieNetwork.getShortTitle(), movieNetwork.getBiggerTitle(), movieNetwork.getMovieTypeId(), movieNetwork.getBanner()))
                .toList()
                .map(Catalog::new);
    }
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/catalog/repo/network/CatalogNetworkMapper.kt
package com.hypnoweb.hypnowebapp.catalog.repo.network

import com.hypnoweb.hypnowebapp.catalog.Catalog
import io.reactivex.Observable
import io.reactivex.Single
import javax.inject.Inject


class CatalogNetworkMapper @Inject constructor() {

    fun fromNetworkCatalog(catalogNetwork: CatalogNetwork): Single<Catalog> {
        return Observable.just(catalogNetwork.movies!!)
                .flatMapIterable { it }
                .map { Catalog.Movie(it.movieId, it.shortTitle, it.biggerTitle, it.movieTypeId, it.banner) }
                .toList()
                .map { Catalog(it) }
    }
}
